File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/window/REHorizontalWindow.java
package io.ganguo.demo.view.window;

import android.content.Context;

import io.ganguo.demo.viewmodel.window.REHorizontalWindowVModel;
import io.ganguo.viewmodel.databinding.IncludeRecyclerPopupwindowBinding;
import io.ganguo.vmodel.ViewModelPopupWindow;

/**
 * Horizontal 方向 - RecyclerView PopupWindow
 * Created by leo on 2018/10/21.
 */
public class REHorizontalWindow extends ViewModelPopupWindow<IncludeRecyclerPopupwindowBinding, REHorizontalWindowVModel> {
    public REHorizontalWindow(Context context) {
        super(context);
    }

    @Override
    public void onViewAttached(REHorizontalWindowVModel viewModel) {

    }

    @Override
    public boolean isWindowWidthFull() {
        return false;
    }

    @Override
    public boolean isWindowHeightFull() {
        return false;
    }

    @Override
    public boolean isTouchOutsideDismiss() {
        return true;
    }

    @Override
    public REHorizontalWindowVModel createViewModel() {
        return new REHorizontalWindowVModel();
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/window/REHorizontalWindow.kt
package io.ganguo.demo.view.window

import android.content.Context

import io.ganguo.demo.viewmodel.window.REHorizontalWindowVModel
import io.ganguo.viewmodel.databinding.IncludeRecyclerPopupwindowBinding
import io.ganguo.viewmodel.core.view.ViewModelWindow

/**
 * Horizontal 方向 - RecyclerView PopupWindow
 * Created by leo on 2018/10/21.
 */
class REHorizontalWindow(context: Context) : ViewModelWindow<IncludeRecyclerPopupwindowBinding, REHorizontalWindowVModel>(context) {


    override fun createViewModel(): REHorizontalWindowVModel {
        return REHorizontalWindowVModel()
    }

    override fun onViewAttached(viewModel: REHorizontalWindowVModel) {
    }

    override val context: Context = context
    override val binding: IncludeRecyclerPopupwindowBinding by lazy {
        viewModel.viewInterface.binding
    }
}
